rc/extra/gdb.kak

```diff
@@ -0,0 +1,413 @@
+                }
+            }
+            /\^done/ {
+                if (printing) {
+                    # trim trailing \n
+                    print_value = substr(print_value, 0, match(print_value, "(\n|\\\\n)*$") - 1)
+                    # QUOTE => \\QUOTE
+                    gsub("'\''", "\\\\'\''", print_value)
+                    # eval -client $client QUOTE info  -- \QUOTE $string \QUOTE QUOTE
+                    send("eval -client '"$kak_client"' '\''info -- \\'\''" print_value "\\'\'\''")
+                    printing = 0
+                }
+            }
+            '
+        } 2>/dev/null >/dev/null &
+        printf "$!" > "${tmpdir}/pid"
+        printf "set-option global gdb_dir %s\n" "$tmpdir"
+    }
+    hook -group gdb global BufOpenFile .* %{
+        gdb-refresh-location-flag %val{buffile}
+        gdb-refresh-breakpoints-flags %val{buffile}
+    }
+    hook -group gdb global KakEnd .* %{
+        gdb-session-stop
+    }
+}
+
+define-command gdb-session-stop %{
+    try %{
+        %sh{
+            if [ -n "$kak_opt_gdb_dir" ]; then
+                kill $(ps -o pid= --ppid $(cat "${kak_opt_gdb_dir}/pid"))
+                rm "${kak_opt_gdb_dir}/pid" "${kak_opt_gdb_dir}/pipe"
+                rmdir "$kak_opt_gdb_dir"
+            else
+                echo raise
+            fi
+        }
+        set-option global gdb_dir ""
+
+        gdb-clear-location
+        gdb-clear-breakpoints
+
+        remove-hooks global gdb
+    }
+}
+
+define-command gdb-jump-to-location %{
+    %sh{
+        if [ -n "$kak_opt_gdb_location_info" ]; then
+            line="${kak_opt_gdb_location_info%%|*}"
+            buffer="${kak_opt_gdb_location_info#*|}"
+            printf "edit -existing \"%s\" %s\n" "$buffer" "$line"
+        fi
+    }
+}
+
+define-command -params 1 gdb-cmd %{
+    %sh{
+        if [ -n "$kak_opt_gdb_dir" ]; then
+            printf %s\\n "$1" > "$kak_opt_gdb_dir"/pipe
+        fi
+    }
+}
+
+define-command gdb-run              %{ gdb-cmd run }
+define-command gdb-start            %{ gdb-cmd start }
+define-command gdb-step             %{ gdb-cmd step }
+define-command gdb-next             %{ gdb-cmd next }
+define-command gdb-finish           %{ gdb-cmd finish }
+define-command gdb-continue         %{ gdb-cmd continue }
+define-command gdb-advance          %{ gdb-cmd "advance %val{buffile}:%val{cursor_line}" }
+define-command gdb-set-breakpoint   %{ gdb-cmd "break %val{buffile}:%val{cursor_line}" }
+define-command gdb-clear-breakpoint %{ gdb-cmd "clear %val{buffile}:%val{cursor_line}" }
+define-command gdb-toggle-breakpoint %{
+    try %{
+        %sh{
+            printf %s\\n "$kak_opt_gdb_breakpoints_info" | tr ':' '\n' |
+            while read -r current; do
+                buffer="${current#*|*|*|}"
+                line=$(printf %s "$current" | cut -d \| -f 3)
+                if [ "$buffer" = "$kak_buffile" ] && [ "$line" = "$kak_cursor_line" ]; then
+                    printf %s\\n raise
```

Comment:
Ah no it doesn't work actually, because `exit` only applies to the subshell, so we end up setting the breakpoint again. Guess I'll go with a heredoc solution